group/assignment5/src/Constraint.java

```diff
@@ -1,36 +1,41 @@
 /**
  * This class represents an order-relation constraint between two Comparable objects from the same type.
- * 'from' is the vertex from which we can go to the second one, 'to'.
- * @param <T>
+ * 'former' is the vertex from which we can go to the second one, 'latter'.
  */
-public class Constraint<T extends Comparable<T>> {
-    public T from;
-    public T to;
+public class Constraint<T> {
```

Comment:
I don't see a good reason T should extend Comparable, so I removed it.

Reply:
If we leave it this way, we need to edit the class comment (remove the word "comparable").